rm-portal/src/main/java/org/ow2/proactive_grid_cloud_portal/rm/client/RMController.java

```diff
@@ -1337,7 +1365,59 @@ public void onFailure(Throwable caught) {
                                  }
 
                                  public void onSuccess(String result) {
-                                     syncCallBack.onSuccess(parseScriptResult(result));
+                                     syncCallBack.onSuccess(parseListScriptResult(result));
+                                 }
+                             });
+    }
+
+    public void executeNodeSourceScript(final String script, final String engine, final String nodeSourceName,
+            final Callback<String, String> syncCallBack) {
+        rm.executeNodeSourceScript(LoginModel.getInstance().getSessionId(),
+                                   script,
+                                   engine,
+                                   nodeSourceName,
+                                   new AsyncCallback<String>() {
+                                       public void onFailure(Throwable caught) {
+                                           String msg = JSONUtils.getJsonErrorMessage(caught);
+                                           LogModel.getInstance()
+                                                   .logImportantMessage("Failed to execute a script " + script +
+                                                                        " on node source " + nodeSourceName + " : " +
+                                                                        msg);
+                                           if (msg.equals("HTTP 500 Internal Server Error")) {
```

Comment:
can you create a separate private method and delegate to it this error management? like handleError(String msg).....

Reply:
Yes, done.